maia/src/protocol.rs

```diff
@@ -329,96 +301,21 @@ pub fn finalize_spend_transaction(
     Ok(tx)
 }
 
-#[derive(Clone, Debug)]
-pub struct PartyParams {
-    pub lock_psbt: PartiallySignedTransaction,
-    pub identity_pk: PublicKey,
-    pub lock_amount: Amount,
-    pub address: Address,
-}
-
-#[derive(Debug, Copy, Clone)]
-pub struct PunishParams {
-    pub revocation_pk: PublicKey,
-    pub publish_pk: PublicKey,
-}
-
-#[derive(Debug, Clone)]
-pub struct CfdTransactions {
-    pub lock: PartiallySignedTransaction,
-    pub commit: (Transaction, EcdsaAdaptorSignature),
-    pub cets: Vec<Cets>,
-    pub refund: (Transaction, Signature),
-}
-
-/// Group of CETs associated with a particular oracle announcement.
-///
-/// All of the adaptor signatures included will be _possibly_ unlocked
-/// by the attestation corresponding to the announcement. In practice,
-/// only one of the adaptor signatures should be unlocked if the
-/// payout intervals are constructed correctly. To check if an adaptor
-/// signature can be unlocked by a price attestation, verify whether
-/// the price attested to lies within its interval.
-#[derive(Debug, Clone)]
-pub struct Cets {
-    pub event: Announcement,
-    pub cets: Vec<(Transaction, EcdsaAdaptorSignature, interval::Digits)>,
-}
-
-#[derive(Debug, Clone, Eq)]
-pub struct Announcement {
-    pub id: String,
-    pub nonce_pks: Vec<schnorrsig::PublicKey>,
-}
-
-impl std::hash::Hash for Announcement {
-    fn hash<H: Hasher>(&self, state: &mut H) {
-        self.id.hash(state)
-    }
-}
-
-impl PartialEq for Announcement {
-    fn eq(&self, other: &Self) -> bool {
-        self.id.eq(&other.id)
-    }
-}
-
-#[derive(Debug, Clone, PartialEq)]
-pub struct Payout {
-    digits: interval::Digits,
+struct PayoutAmounts {
     maker_amount: Amount,
     taker_amount: Amount,
 }
 
-pub fn generate_payouts(
-    range: RangeInclusive<u64>,
-    maker_amount: Amount,
-    taker_amount: Amount,
-) -> Result<Vec<Payout>> {
-    let digits = interval::Digits::new(range).context("invalid interval")?;
-    Ok(digits
-        .into_iter()
-        .map(|digits| Payout {
-            digits,
-            maker_amount,
-            taker_amount,
-        })
-        .collect())
-}
-
-impl Payout {
-    pub fn digits(&self) -> &interval::Digits {
-        &self.digits
-    }
-
-    pub fn maker_amount(&self) -> &Amount {
-        &self.maker_amount
-    }
-
-    pub fn taker_amount(&self) -> &Amount {
-        &self.taker_amount
+impl From<Payout> for PayoutAmounts {
+    fn from(payout: Payout) -> Self {
+        Self {
+            maker_amount: *payout.maker_amount(),
+            taker_amount: *payout.taker_amount(),
+        }
```

Comment:
🙃 It's a bit harder to review the commit which creates the new crate if it also includes adding new types and methods.

Reply:
I agree, but this was the only way out of this. the very thing we want to modify would have otherwise go to `maia-core`, and then we wouldn't be able to change it :/

In hindsight, I could have tried to extract commit that breaks up `Payout`.
Can still do it if you think it would be easier.

Reply:
I think it's clear enough in this case, just a comment for the future!